### PR TITLE
[fix][calendar] Fix iOS 17 permissions in current module

### LINKED ISSUE #24343: iOS 17 support for calendar permissions scope: `requestAccessToEntityType` deprecated in favor of `requestFullAccessToEventsWithCompletion`
### Minimal reproducible example

https://snack.expo.dev/hpDQeKGu7

### Summary

In iOS 17, they have broken out the permissions scope to include Read, write, and full access where previously it was only full access.  As a result the current `Calendar.requestCalendarPermissionsAsync()` throws the following error when run on iOS 17, but works on iOS 16.

`Error: -requestAccessToEntityType:completion: has been deprecated-calling this method is no longer allowed. Instead, use -requestFullAccessToEventsWithCompletion:, -requestWriteOnlyAccessToEventsWithCompletion:, or -requestFullAccessToRemindersWithCompletion:.`

See deprecation  notice for  `requestAccessToEntityType`  here: https://developer.apple.com/documentation/eventkit/ekeventstore/1507547-requestaccesstoentitytype

See here for more info: https://developer.apple.com/documentation/eventkit/ekeventstore/4162272-requestfullaccesstoeventswithcom

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.4.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.19.0 - ~/.nodenv/versions/16.19.0/bin/node
      Yarn: 1.22.19 - ~/.nodenv/versions/16.19.0/bin/yarn
      npm: 8.19.3 - ~/.nodenv/versions/16.19.0/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, visionOS 1.0, watchOS 10.0
      Android SDK:
        API Levels: 24, 26, 28, 29
        Build Tools: 23.0.2, 24.0.1, 26.0.0, 26.0.2, 27.0.3, 28.0.2, 28.0.3, 29.0.2
        System Images: android-26 | Google APIs Intel x86 Atom, android-28 | Google Play Intel x86 Atom, android-29 | Google APIs Intel x86 Atom
    IDEs:
      Android Studio: 3.6 AI-192.7142.36.36.6392135
      Xcode: 15.0/15A5195m - /usr/bin/xcodebuild
    npmGlobalPackages:
      expo-cli: 6.0.8

### PR DESCRIPTION
**Pull request by alanjhughes**
# Why
Closes #24343

In case we want to hold off on the full expo modules migration #24282 until SDK 50 this addresses the iOS permissions error in the current module

# How
Use the new method to request permissions if on iOS 17. Added new keys to the plist and updated the config plugin

# Test Plan
bare-expo. Passes on iOS 16 and 17

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]

File: ios/Podfile.lock [hunk 1]
```diff
@@ -4975,7 +4975,7 @@ SPEC CHECKSUMS:
   EXUpdates: a910a0b964a12f9c7a6d673bad3c5eb70249fc5e
   EXUpdatesInterface: 98f69982f4b429eb62a1d99259bc980038240f27
   FBAEMKit: af2972f39bb0f3f7c45998f435b007833c32ffb2
-  FBAudienceNetwork: 4d6223a9a88a22e9dca8c9e6c1829a8745b86228
+  FBAudienceNetwork: 03e273f66b70756be7d060927111af97b7641b06
```
[the remark this answers is not in the document]
Comment (alanjhughes) [reply ios/Podfile.lock:4978]: Yeah, that happened after a rebase but not sure where it's coming from 🤔

[the remark this answers is not in the document]
Comment (Kudo) [reply ios/Podfile.lock:4978]: updated in this commit already https://github.com/expo/expo/commit/94a58ec8b9061f9b3e66bccc918ccddaab48c3e3

we could confirm the `03e273f66b70756be7d060927111af97b7641b06` is correct one

File: .github/workflows/client-ios.yml [hunk 2]
```diff
@@ -42,8 +42,8 @@ jobs:
         uses: actions/checkout@v3
         with:
           submodules: true
-      - name: 🔨 Switch to Xcode 14.3
-        run: sudo xcode-select --switch /Applications/Xcode_14.3.app
+      - name: 🔨 Switch to Xcode 15.0
+        run: sudo xcode-select --switch /Applications/Xcode_15.0.app
```
Comment (Kudo) [inline .github/workflows/client-ios.yml:46]: by updating xcode you should probably revert #23874, but i wonder that would make test suite ios ci more unstable. would need further figuring out how long it takes to boot up ios 17 simulator.

Comment (alanjhughes) [reply .github/workflows/client-ios.yml:46]: I'll do that and see how long it takes 👍

**Review by lukmccall — APPROVED** [review]

**Pull request merged**